### PR TITLE
Unified Pipelines for Java Repo

### PR DESCRIPTION
**Pull request by mitchdenny**
This change lands the underlying changes for the Unified Pipelines work in the Java mono-repo. The key changes are:

1. Converted the job template to a stage template (relocated to a different folder).
2. Added a conditional template reference to the end which is used for CI builds in internal.
3. Modifies all ```ci.yml``` files to include a reference to the build tools repo, and explicitly specify their artifacts.

Additionally I've set a parameter on the template called ```EnableUnifiedPipelines```. This parameter controls the activation of the extended pipeline on a per service directory basis so that we can opt service directories in as we are ready.

### CODE REVIEW

**Comment by mitchdenny** [comment]
@danieljurek we probably need to talk about how this intersects with the ```test.yml``` files - and @g2vinay we need to talk about the ```sdk/cosmos``` folder as I was unable to retrofit that one since its ```ci.yml``` is non-standard (this needs to be fixed).

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+      strategy:
+        matrix:
+          Linux - Java 8:
+            OSName: 'Linux'
+            OSVmImage: 'ubuntu-16.04'
+            JavaVersion: '1.8'
+            ProfileFlag: ''
+          macOS - Java 8:
+            OSName: 'macOS'
+            OSVmImage: 'macOS-10.13'
```
Comment (joshfree) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml]: is upgrading to 10.14 images on our radar soon?

**Comment by mitchdenny** [comment]
Going to abandon this PR due to some complex merge issues that we are going to have, going to be easier to just make my tweaks again.

File: eng/pipelines/templates/stages/archetype-sdk-client.yml (hunk 1 above)
Comment (mitchdenny) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml]: We didn't have anything planned yet, but we can if there is a need?

**Comment by mitchdenny** [comment]
Closing this PR down in favor of this one #4807

File: eng/pipelines/templates/stages/archetype-sdk-client.yml (hunk 1 above)
Comment (danieljurek) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml]: We do have a work item to update Windows build agents: https://github.com/azure/azure-sdk/issues/527

I've opened a work item to upgrade to 10.14: https://github.com/azure/azure-sdk/issues/527

**Pull request closed without merging**